### DISCUSSION TITLE
Adding link back to table of contents

### DISCUSSION THREAD

**Discussion by dannyduncan** [executablebooks/meta#167] [Q&A]
Within the jupyterbook framework, I'm attempting to to add a link back to the table of contents after rendering using latex. This is not really working for me using standard internal linking methods [name of link](name of .md or .ipynb document)

Much appreciate any help that anyone can provide.

**Comment by choldgraf** [comment]
ah so you wish to have a link to the table of contents within the PDF itself? Perhaps @mmcky has experience with this?

**Reply by dannyduncan** [reply]
@choldgraf yes that's correct!

**Comment by mmcky** [comment]
hi @dannyduncan -- I just tried adding

```md
[toc](intro.md)
```

for a project where the first `- file: intro` is specified in the `_config.yml` and it links to the bottom of that page in `latex` and not particularly useful.

In `sphinx` (which is what we use under the hood to build the html) there is the option to use `:name:` on the `toctree` directive for use with `ref`. @choldgraf perhaps we should add a default `label` name for the `global toc` that could make it able to be referenced?

**Reply by mmcky** [reply]
@choldgraf I will try adding

```
:name: mastertoc
```

to

https://github.com/executablebooks/jupyter-book/blob/a0109e54d8ae415081b43181286b651a3c963929/jupyter_book/toc.py#L203

**Reply by mmcky** [reply]
or perhaps something else without `master` is a better idea

```
:name: globaltoc
```

is a better description.

**Reply by dannyduncan** [reply]
@mmcky yes I think adding the name would make referencing possible. Please let me know if there's anything I can help with here.

**Reply by mmcky** [reply]
[this pr](https://github.com/executablebooks/jupyter-book/pull/1060) looks to resolve this issue by adding a `globaltoc` label for the `masterdoc` -- however the placement of the `toctree` in the xml is currently an issue linking to the bottom of content placed in the first document of a project. 

Current `latex` output

```latex
...
\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{intro::doc}}


This is the frontmatter

\phantomsection\label{\detokenize{intro:globaltoc}}
...
```

but needs to be:

```latex
\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\phantomsection\label{\detokenize{intro:globaltoc}}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{intro::doc}}


This is the frontmatter
```

**Reply by mmcky** [reply]
> @mmcky yes I think adding the name would make referencing possible. Please let me know if there's anything I can help with here.

@dannyduncan unfortunately it seems that adding a `:name:` is ignored by the sphinx latex writer. I guess because it relies on latex to generate the table of contents.

**Reply by dannyduncan** [reply]
@mmcky that's unfortunate but thank you for trying to accomplish a workaround!

**Reply by mmcky** [reply]
@dannyduncan I have noted this a feature request and will add it to our work on LaTeX templates and making improvements to the PDF output.
